src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6693,10 +6693,16 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         }
         this->visit_expr(*x.m_value);
 
-        // If tmp is a statement and not an expression
-        // never cast into expression using ASRUtils::EXPR
-        // Just ignore and exit the function naturally.
-        if( tmp && !ASR::is_a<ASR::stmt_t>(*tmp) ) {
+        if (eval_count > 0) {
+            // In Interactive mode
+            if ((tmp) && (!ASR::is_a<ASR::expr_t>(*tmp))) {
+                LCOMPILERS_ASSERT(ASR::is_a<ASR::expr_t>(*tmp));
+                tmp = nullptr;
```

Comment:
I did not quite get the logic here. It says `if tmp exists and it is not an expression`, then right inside the `if`, we have `assert that tmp is an expression`. `tmp is not an expresssion` in `if` and `tmp is an expression` in the assert are contradicting, right?

Reply:
Yes. It does not make sense. I should remove that assertion.